### PR TITLE
Shortcuts

### PR DESCRIPTION
**Pull request by Daio-io**
- Added support for keyboard shortcuts 
- Refactor modals to better handle lifecycle with a base Modal class
- Created a help modal to display keyboard shortcut options
- Added keyboard css styling to show shortcut keys in views
- Added shortcuts to language select nav and saving

### CODE REVIEW

**Comment by alechewitt** [comment]
Nice all looks good to me. I like the keyboard class you made!
However think we should also detect if they are on mac or windows, and then display cmd/ctrl accordingly. But this can be another ticket.

File: resources/styles/main.less [hunk 1]
```diff
@@ -6,58 +6,82 @@
...
+  @media all and (max-width: 465px) {
+    display: none;
+  }
 }
 
 .menu-items li:hover {
-    cursor             : pointer;
-    -webkit-box-shadow : 0px 3px 0px #E64002 inset;
-    -moz-box-shadow    : 0px 3px 0px #E64002 inset;
-    box-shadow         : 0px 3px 0px #E64002 inset;
```
Comment (alechewitt) [inline resources/styles/main.less:52]: Are you using some kind of auto spacing here?, Generally with CSS I like to line it all up, just to make it a bit easier to read, but we don't have to.
I have just realised we have ESLint configuration file but it is not running as part of Gulp. We should get that running by default every time we push, and in the develop task.

**Comment by Daio-io** [comment]
Yeah, I already created a ticket for that and left the work because its something that would work nicer when we have templating in place to update the text to cmd or Ctrl based on platform :)

File: resources/styles/main.less (hunk 1 above)
Comment (Daio-io) [inline resources/styles/main.less:52]: It's just intellij formatting so feel free to clean that up and add linting

**Comment by alechewitt** [comment]
Yep your right, we need proper templating in place before doing that

**Pull request merged**